client-go/src/main/client/session.go

```diff
@@ -0,0 +1,698 @@
+	dir, _ := os.Getwd()
+	os.Mkdir(dir+"\\logs", os.ModePerm)
+	logFile, _ := os.OpenFile(dir+"\\logs\\all.log", os.O_CREATE|os.O_RDWR|os.O_APPEND, 0666)
+	Log.SetOutput(logFile)
+	var protocolFactory thrift.TProtocolFactory
+	s.trans, s.err = thrift.NewTSocketTimeout(net.JoinHostPort(s.Host, s.Port), time.Duration(connectionTimeoutInMs))
+	if s.err != nil {
+		Log.WithError(s.err).Error("connect failed")
+	}
+	s.trans = thrift.NewTFramedTransport(s.trans)
+	if !s.trans.IsOpen() {
+		s.err = s.trans.Open()
+		if s.err != nil {
+			Log.WithError(s.err).Error("open the conn failed")
+		}
+	}
+	if enableRPCCompression {
+		protocolFactory = thrift.NewTCompactProtocolFactory()
+	} else {
+		protocolFactory = thrift.NewTBinaryProtocolFactoryDefault()
+	}
+	iProtocol := protocolFactory.GetProtocol(s.trans)
+	oProtocol := protocolFactory.GetProtocol(s.trans)
+	s.client = rpc.NewTSIServiceClient(thrift.NewTStandardClient(iProtocol, oProtocol))
+	s.ZoneId = DefaultZoneId
+	tSOpenSessionReq := rpc.TSOpenSessionReq{ClientProtocol: protocolVersion, ZoneId: s.ZoneId, Username: &s.User,
+		Password: &s.Passwd}
+	tSOpenSessionResp, err := s.client.OpenSession(context.Background(), &tSOpenSessionReq)
+	if err != nil {
+		Log.WithError(err).Error("open session failed")
+	} else {
+		Log.WithField("code", tSOpenSessionResp.GetStatus().Code).Info("open session success")
+	}
+	s.sessionId = tSOpenSessionResp.GetSessionId()
+	s.requestStatementId, err = s.client.RequestStatementId(context.Background(), s.sessionId)
+	s.FetchSize = DefaultFetchSize
+	if err != nil {
+		Log.WithError(err).Error("request StatementId failed")
+	}
+	if s.ZoneId != "" {
+		s.SetTimeZone(s.ZoneId)
+	} else {
+		s.ZoneId = s.GetTimeZone()
+	}
+}
+
+func (s *Session) CheckTimeseriesExists(path string) bool {
+	dataSet := s.ExecuteQueryStatement("SHOW TIMESERIES " + path)
+	result := dataSet.HasNext()
+	dataSet.CloseOperationHandle()
+	return result
+}
+
+func (s *Session) Close() error {
+	tSCloseSessionReq := rpc.NewTSCloseSessionReq()
+	tSCloseSessionReq.SessionId = s.sessionId
+	status, err := s.client.CloseSession(context.Background(), tSCloseSessionReq)
+	if err != nil {
+		Log.WithError(err).Error("Error occurs when closing session at server. Maybe server is down. Error message")
+		return err
+	} else {
+		Log.WithField("code", status.Code).Info("close session success")
+	}
+	s.trans.Close()
+	return nil
+}
+
+/*
+ *set one storage group
+ *
+ *param
+ *storageGroupId: string, storage group name (starts from root)
+ *
+ *return
+ *error: correctness of operation
+ */
+func (s *Session) SetStorageGroup(storageGroupId string) error {
+	status, err := s.client.SetStorageGroup(context.Background(), s.sessionId, storageGroupId)
+	if err != nil {
+		Log.WithError(err).Error("setting storage group failed")
+	} else {
+		Log.WithFields(logrus.Fields{
+			"sg":   storageGroupId,
+			"code": status.Code,
+		}).Info("setting storage group success")
```

Comment:
I think if the operation succeeds, we do not need to print the info message, as this will print so many logs for users who use the go-client. I think the debug message is ok. and the default log level is info.